src/System.Drawing.Common/tests/Drawing2D/GraphicsPathTest.cs

```diff
@@ -0,0 +1,3294 @@
+        [InlineData(1, 1, 2, 2)]
+        public void AddLines_Success(int x1, int y1, int x2, int y2)
+        {
+            using (GraphicsPath gpi = new GraphicsPath())
+            using (GraphicsPath gpf = new GraphicsPath())
+            {
+                gpi.AddLines(new Point[] { new Point(x1, y1), new Point(x2, y2) });
+                AssertLine(gpi);
+
+                gpf.AddLines(new PointF[] { new PointF(x1, y1), new PointF(x2, y2) });
+                AssertLine(gpf);
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(1, 1)]
+        public void AddLines_SinglePoint_Success(int x1, int y1)
+        {
+            using (GraphicsPath gpi = new GraphicsPath())
+            using (GraphicsPath gpf = new GraphicsPath())
+            {
+                gpi.AddLines(new PointF[] { new PointF(x1, y1) });
+                AssertLinesSinglePoint(gpi);
+
+                gpf.AddLines(new PointF[] { new PointF(x1, y1) });
+                AssertLinesSinglePoint(gpf);
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(49, 157)]
+        public void AddLines_SamePoint_Success(int x1, int y1)
+        {
+            Point[] intPoints = new Point[]
+            {
+                new Point(x1, y1), new Point(x1, y1)
+            };
+
+            PointF[] floeatPoints = new PointF[]
+            {
+                new PointF((float)x1, (float)y1), new PointF((float)x1, (float)y1),
+                new PointF((float)x1, (float)y1), new PointF((float)x1, (float)y1)
+            };
+
+            using (GraphicsPath gpi = new GraphicsPath())
+            using (GraphicsPath gpf = new GraphicsPath())
+            {
+                gpi.AddLines(intPoints);
+                Assert.Equal(2, gpi.PointCount);
+                Assert.Equal(0, gpi.PathTypes[0]);
+                Assert.Equal(1, gpi.PathTypes[1]);
+
+                gpi.AddLines(intPoints);
+                // 3 not 4 points, the first point (only) is compressed
+                Assert.Equal(3, gpi.PointCount);
+                Assert.Equal(0, gpi.PathTypes[0]);
+                Assert.Equal(1, gpi.PathTypes[1]);
+                Assert.Equal(1, gpi.PathTypes[2]);
+
+                gpi.AddLines(intPoints);
+                // 4 not 5 (or 6) points, the first point (only) is compressed
+                Assert.Equal(4, gpi.PointCount);
+                Assert.Equal(0, gpi.PathTypes[0]);
+                Assert.Equal(1, gpi.PathTypes[1]);
+                Assert.Equal(1, gpi.PathTypes[2]);
+                Assert.Equal(1, gpi.PathTypes[3]);
+
+                gpf.AddLines(floeatPoints);
+                // all identical points are added
+                Assert.Equal(4, gpf.PointCount);
+                Assert.Equal(0, gpf.PathTypes[0]);
```

Comment:
Could use `Assert.Equal(IEnumerable<T>, IEnumerable<T>)` here and elsewhere (I'll stop commenting)

e.g
```cs
Assert.Equal(new int[] { 1, 1, 1 }, gpf.PathTypes);
```